test/integ/change_password_integ_test.js

```diff
@@ -0,0 +1,390 @@
+                            "newPassword": "dsfds",
+                            "newPasswordConfirm": "Qwerty123!"
+                        })
+                        .then(async (res) => {
+                            assert.equal(res.body.errorID, "passwordsDoNotMatch");
+                            await checkPassword(TEST_USER, "test");
+                            resolve();
+                        })
+                        .catch((err) => {
+                            reject(err);
+                        });
+                })
+                .catch((err) => {
+                    reject(err);
+                });
+        });
+    });
+    it("should prevent user from changing password if new password is not strong enough", () => {
+        return performAsyncTest((resolve, reject) => {
+            performUserLogin()
+                .then(async (agent) => {
+                    await checkPassword(TEST_USER, "test");
+                    agent.post("/change_password?type=json")
+                        .type("form")
+                        .send({
+                            "oldPassword": "test",
+                            "newPassword": "weak",
+                            "newPasswordConfirm": "weak"
+                        })
+                        .then(async (res) => {
+                            assert.equal(res.body.errorID, "passwordNotStrong");
+                            await checkPassword(TEST_USER, "test");
+                            resolve();
+                        })
+                        .catch((err) => {
+                            reject(err);
+                        });
+                })
+                .catch((err) => {
+                    reject(err);
+                });
+        });
+    });
+    it("should prevent user from changing password if user doesn't exist", () => {
+        return performAsyncTest((resolve, reject) => {
+            performUserLogin()
+                .then(async (agent) => {
+                    await checkPassword(TEST_USER, "test");
+                    usersCollection = db.collection("users");
+                    usersCollection.remove({ username: TEST_USER }, true)
+                        .then((result) => {
+                            agent.post("/change_password?type=json")
+                                .type("form")
+                                .send({
+                                    "oldPassword": "test",
+                                    "newPassword": "something",
+                                    "newPasswordConfirm": "something"
+                                })
+                                .then((res) => {
+                                    assert.equal(res.statusCode, 404);
+                                    checkPassword(TEST_USER, "test")
+                                        .then((res) => {
+                                            assert.fail("Password shouldn't change");
```

Comment:
This failure is swallowed by the `catch` handler below. Other tests have it too.

Reply:
Should be fixed now.